### PR TITLE
Fibonacci

### PR DESCRIPTION
**Pull request by piezol**

### CODE REVIEW

File: module1/homework/fibonacci/fibonacci.hpp [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+        fib_n = temporary_sum;
+    }
+    return fib_n;
 }
 
 int fibonacci_recursive(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    if (sequence == 0) return 0;
+    if (sequence == 1) return 1;
```
Comment (jakjan95) [inline module1/homework/fibonacci/fibonacci.hpp:17]: You can simplify it by making just one condition : sequence <=1 and inside it returning a sequence.

Comment (piezol) [reply module1/homework/fibonacci/fibonacci.hpp:17]: I would not call it 'simplify', as fibonacci sequence is explicitly defined as (F0 = 0, F1 = 1), and it 'just so happens' that it is logically equivalent to (Fn = n for every non-negative integer <= 1), but it's not a natural way humans look at the sequence.

(i assume we don't go to asm and compare which one is better - yes, yours is usually better, but performance was not mentioned).

Comment (nauka-programowania-MA) [reply module1/homework/fibonacci/fibonacci.hpp:17]: I agree with jakjan95 that sequence <=1 is better option.

**Comment by ziobron** [comment]
{ "requirement_ids": [1119, 1430], "points": [2, 6] }

**Pull request closed without merging**